nupic/frameworks/opf/metrics.py

```diff
@@ -484,12 +491,51 @@ def _compute(self):
     return self.aggregateError
 
 
+class MetricNegLogLikelihood(AggregateMetric):
+  """
+      computes negative log-likelihood
```

Comment:
I think it would be good to add some descriptive comments and don't assume they know what this is. When should the user use this metric? Why is this metric better? Perhaps a reference to a wikipedia page or article somewhere.